client/network/src/behaviour.rs

```diff
@@ -343,12 +344,18 @@ Behaviour<B, H> {
 				self.events.push_back(BehaviourOut::NotificationsReceived { remote, messages });
 			},
 			CustomMessageOutcome::PeerNewBest(peer_id, number) => {
-				self.light_client_handler.update_best_block(&peer_id, number);
+				self.light_client_request_sender.update_best_block(&peer_id, number);
+			}
+			CustomMessageOutcome::SyncConnected(peer_id) => {
+				// TODO: Should this be tied to sync connections, or to connections in general?
+				self.light_client_request_sender.inject_connected(peer_id);
+				self.events.push_back(BehaviourOut::SyncConnected(peer_id))
+			}
+			CustomMessageOutcome::SyncDisconnected(peer_id) => {
+				// TODO: Should this be tied to sync connections, or to connections in general?
+				self.light_client_request_sender.inject_disconnected(peer_id);
+				self.events.push_back(BehaviourOut::SyncDisconnected(peer_id))
 			}
```

Comment:
Two arguments for using nodes in the `sync` peer set:
- Simplicity
- Likely long-lived, thus good for requests

Reply:
Sounds OK to me, but maybe @tomaka has a more educated opinion on the matter, being intimately familiar with the peer sets.

Reply:
@tomaka any thoughts on this?

Reply:
Sorry, forgot to answer.
Yes, that is the right thing to do. Being "sync connected" proves that they're on the correct chain.